internal/integrations/v4/executor/executor_windows.go

```diff
@@ -4,11 +4,59 @@ package executor
 
 import (
 	"context"
+	"fmt"
+	"golang.org/x/sys/windows"
 	"os/exec"
+	"unsafe"
 )
 
 // userAwareCmd returns a cancellable Cmd struct to execute the given command with the provided
 // arguments.
 func (r *Executor) userAwareCmd(ctx context.Context) *exec.Cmd {
 	return exec.CommandContext(ctx, r.Command, r.Args...)
 }
+
+// startProcess starts and sets priority to command
+func startProcess(cmd *exec.Cmd) error {
+	err := cmd.Start()
+	if err != nil {
+		return err
+	}
+	err = setPriorityClass(cmd)
+	if err != nil {
+		illog.WithError(err).WithField("command", cmd.String()).Error("cannot set priority class to process")
+	}
+
+	return nil
+}
+
+// setPriorityClass will set the priorityClass of the agent to the cmd process
+func setPriorityClass(cmd *exec.Cmd) error {
+	priorityClass, err := windows.GetPriorityClass(windows.CurrentProcess())
```

Comment:
It's a good concern :) But in this case it cannot happen. The priority is set before "running the agent" 
[Priority set](https://github.com/newrelic/infrastructure-agent/blob/fe014451fed7ce3e2ff98dc3faa1377682f8677d/cmd/newrelic-infra/newrelic-infra.go?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L189) in a [synchronous way ](https://github.com/newrelic/infrastructure-agent/blob/fe014451fed7ce3e2ff98dc3faa1377682f8677d/cmd/newrelic-infra/initialize/initialize_windows.go#L61)

What I'm not sure if it could happen is that somehow failed changing the agent priority. (But this would be a problem per se so I wouldn't go that far)

Still, ping me if you think it deserves further analysis/discussion  :)